### PR TITLE
Docs: Fluid dropdown

### PR DESCRIPTION
**Pull request by aagonzales**
Closes https://github.com/carbon-design-system/carbon-website/issues/3174
Closes https://github.com/carbon-design-system/carbon-website/issues/3173
Closes https://github.com/carbon-design-system/carbon-website/issues/3172

- Docs for fluid dropdown, combobox, multiselect
- "Fluid input" branch updates from "main" (or have I missed something up? lol)

#### Changelog

*Usage*
- Add a "Styling" section under Formatting to give an overview of Fixed and Fluid styles 
   - This is different than how I did it in Text Input, where this section was under each variant, if we like this way better can change text input to match later.
- Add `tab` component with Fluid image example to select images (not all have one)
- Mover "disabled" and "invalid" sections previously under "Modifiers" to a "States" section under "Universal Behaviors"
- Removed `light` prop section under modifiers.

*Style*
- Add tab component and fluid images to color section
- Add a Fluid dropdown structure section
   - I didn't add a new fluid section to Combobox or MultiSelect as the page would get very long. My thinking is that the structure for those can be inferred from their fixed version and the fluid default dropdown. Do you agree? Or should I go ahead and add those sections too?
- Move "Size" section to the bottom
- Added a fluid size section

**Testing/Reviewing**
- Go to the "Dropdown" page and proof read add fluid sections and images on usage and style tabs.

### CODE REVIEW

**Comment by tw15egan** [comment]
@aagonzales I'm going to push up a PR that updates the `fluid-input` branch with the latest from `main` so this PR is cleaner

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Looks great! Just a couple of things:

### Usage tab

**Multiselect > Second image**
- Something may be off in the second fixed/fluid image for Multiselect. The tag shifts from left to right between viewing fixed/fluid. I would double-check it to see if there is a spacing issue in one of them. 
dropdown

**Multiselect > Filtering section images**
First image: The “Data flows” option should be `$text-secondary`.
Screen Shot 2022-10-17 at 4 31 38 PM

Second image: I think the “Data visualizations” option should also be `$text-secondary` here since all search results have the same importance before hovering.
Screen Shot 2022-10-17 at 4 32 46 PM

-------

### Style tab
**Interactions > Combobox fluid image:**
- The Warning state message should be `$text-primary`.

Screen Shot 2022-10-17 at 4 49 17 PM

- The Item hover state is missing the cursor like you have shown in the fixed image.

**Sizes > Fluid image**
- The image size looks enlarged within the space. Also seems like the fixed image size is too small.

File: src/pages/components/dropdown/style.mdx [hunk 1]
```diff
@@ -159,33 +200,35 @@ various color differences for closed and open dropdowns.
...
-| Menu option | Small (sm)  | 32 / 2          |
-|             | Medium (sm) | 40 / 2.5        |
-|             | Large (lg)  | 48 / 3          |
+| Element      | Property                    | px / rem   | Spacing token |
+| ------------ | --------------------------- | ---------- | ------------- |
+| Label        | margin-bottom               | 4 / 0.25   | `$spacing-02` |
+| Field        | padding-left                | 16 / 1     | `$spacing-05` |
+|              | padding-right               | 48 / 3     | `$spacing-09` |
+|              | border-bottom               | 1px        | –             |
+|              | margi-top, margin-bottom    | 13 / .8125 | –             |
```
Comment (laurenmrice) [inline src/pages/components/dropdown/style.mdx:219]:
```suggestion
|              | margin-top, margin-bottom    | 13 / .8125 | –             |
```

File: src/pages/components/dropdown/style.mdx [hunk 2]
```diff
@@ -243,3 +286,50 @@ can be applied to all variants of dropdown.
 </div>
 
 <Caption>Structure and spacing for inline dropdown | px / rem</Caption>
+
+## Sizes
+
+### Fixed inputs
+
+The field height and menu option height should always match. These sizes options
```
Comment (laurenmrice) [inline src/pages/components/dropdown/style.mdx:294]:
```suggestion
The field height and menu option height should always match. These size options
```

File: src/pages/components/dropdown/style.mdx [hunk 3]
```diff
@@ -243,3 +286,50 @@ can be applied to all variants of dropdown.
...
+can be applied to all variants of dropdown.
+
+| Element   | Size        | Height px / rem |
+| --------- | ----------- | --------------- |
+| Field     | Small (sm)  | 32 / 2          |
+|           | Medium (md) | 40 / 2.5        |
+|           | Large (lg)  | 48 / 3          |
+| Menu item | Small (sm)  | 32 / 2          |
+|           | Medium (sm) | 40 / 2.5        |
```
Comment (laurenmrice) [inline src/pages/components/dropdown/style.mdx:303]:
```suggestion
|           | Medium (md) | 40 / 2.5        |
```

File: src/pages/components/dropdown/usage.mdx [hunk 4]
```diff
@@ -354,10 +425,11 @@ dropdowns open downward.
 
 ### Elevation
 
-Carbon has one layer style called a `box-shadow`. The box-shadow is placed
-behind the dropdown menu when open. Box-shadow is also used in other components
-that have overlayed menus, such as the overflow menu and date picker calendar.
-The SCSS for box-shadow is `0 2px 6px 0 rgba(0,0,0,.2)`.
+Carbpn rarely uses The dropdown menu has a style called `box-shadow` that is
```
Comment (laurenmrice) [inline src/pages/components/dropdown/usage.mdx:428]:
```suggestion
The dropdown menu has a style called `box-shadow` that is
```

**Review by laurenmrice — CHANGES_REQUESTED** [review]

**Review by laurenmrice — APPROVED** [review]

**Comment by aagonzales** [comment]
@tw15egan it's not showing any conflict anymore, can we just merge this one? Or should I still open a new clean one?

**Comment by tw15egan** [comment]
As long as everything looks good on your end, I'm good with merging it 👍🏻

**Review by tw15egan — APPROVED** [review]

**Pull request merged**